packages/experimental/Drawer/src/Drawer.types.ts

```diff
@@ -0,0 +1,221 @@
+   * */
+  handleBackgroundColor?: ColorValue;
+
+  /**
+   * The corner radius of the handle.
+   * */
+  handleCornerRadius?: number;
+
+  /**
+   * The top margin of the handle.
+   *  */
+  handleMarginTop?: number;
+
+  /**
+   * The bottom margin of the handle.
+   * */
+
+  handleMarginBottom?: number;
+
+  /**
+   * The start margin of the handle.
+   * */
+
+  handleMarginStart?: number;
+
+  /**
+   * The end margin of the handle.
+   * */
+
+  handleMarginEnd?: number;
+
+  /**
+   * The flex alignment of the handle.
+   * */
+  handleAlignment?: FlexAlignType;
+
+  /**
+   * The background for drawer
+   * */
+
+  drawerBackgroundColor?: ColorValue;
+
+  /**
+   * The corner radius of the Drawer
+   * @default 0
+   * */
+  drawerCornerRadius?: number;
+
+  /**
+   * The elevation of the Drawer
+   * @default 5
+   * @platform android
+   * */
+  drawerElevation?: number;
+
+  /**
+   * The width of the Drawer
+   * Note: Only applicable when position is 'left' or 'right'
+   * @default '100%'
+   * */
+  width?: number | string;
+
+  /**
+   * The height of the Drawer
+   * Note: Only applicable when position is 'bottom'
+   * @default '40%'
+   * */
+  height?: number | string;
+
+  /**
+   * The shadow color of the Drawer
+   * @platform iOS
+   * */
+  shadowColor?: ColorValue;
+
+  /**
+   * The shadow offset of the Drawer
+   * @platform iOS
+   * */
+  shadowOffset?: { width: number; height: number };
+
+  /**
+   * The shadow opacity of the Drawer
+   * @platform iOS
+   * */
+  shadowOpacity?: number;
+
+  /**
+   * The shadow radius of the Drawer
+   * @platform iOS
+   * */
+
+  shadowRadius?: number;
+  /**
+   * The behavior of the Drawer
+   * @default 'leftSlideOver'
+   * */
+  behavior?: DrawerBehaviorType;
+
+  /**
+   * Positions/Behaviours of the Drawer
+   */
+
+  leftSlideOver?: DrawerTokens;
+  rightSlideOver?: DrawerTokens;
+  bottomSlideOver?: DrawerTokens;
+  bottom?: DrawerTokens;
+
+  /**
+   * Coordinate position of the Drawer
+   */
+
+  leftPosition?: number;
+  rightPosition?: number;
+  bottomPosition?: number;
+  topPosition?: number;
+}
+
+export interface DrawerProps extends PressableFocusProps {
+  /*
+   ** An accessibility label for screen readers. Set on the text Drawer.
+   */
+  accessibilityLabel?: string;
+
+  /**
+   * Visibility of the Drawer
+   */
+  visible?: boolean;
+
+  /**
+   * Callback when the Drawer is closed
+   * */
+  onClose?: (e: InteractionEvent) => void;
+
+  /**
+   * Callback when the Drawer is opened
+   * */
+  onOpen?: (e: InteractionEvent) => void;
+
+  /**
+   * Whether the Drawer is open on mount
+   */
+  defaultOpen?: boolean;
+
+  /**
+   * Callback when the scrim is clicked
+   * */
+  onScrimClick?: (e: InteractionEvent) => void;
+
+  /**
+   * The content of the Drawer
+   * */
+  children?: React.ReactNode;
+
+  /**
+   * The behavior of the Drawer
+   * @default 'leftSlideOver'
+   * */
+  behavior?: DrawerBehaviorType;
```

Comment:
nit: drawerPosition prop with values 'left' 'right' 'bottom' seems more intuitive imo

Reply:
We tried to match the native naming convention with design suggested. 
But let me go through once and maybe we can remove slideOver and change the name of behaviour that is different than normal one.

Reply:
I think this is good suggestion on the naming. 
Changing to drawerPosition with value only to 'left' 'right' 'bottom'  , while bottom drawer that have extra behaviour which can be take care by another prop if passed true.